### PR TITLE
Refine settings driver

### PR DESCRIPTION
**Pull request by chshu**
- write the actual setting's length into flash

  - add getAlignLength() for address alignment

### CODE REVIEW

File: src/core/common/settings.cpp [hunk 1]
```diff
@@ -345,6 +351,13 @@ ThreadError otPlatSettingsGet(otInstance *aInstance, uint16_t aKey, int aIndex,
 
 ThreadError otPlatSettingsSet(otInstance *aInstance, uint16_t aKey, const uint8_t *aValue, int aValueLength)
 {
+    int length;
+
+    if (otPlatSettingsGet(aInstance, aKey, 0, NULL, &length) == kThreadError_None)
+    {
+        otPlatSettingsDelete(aInstance, aKey, 0);
```
Comment (jwhui) [inline src/core/common/settings.cpp]: We should not be deleting before writing.  For example, if the device resets after the delete but before the write, there will be no value for this setting after reboot.

We should not need to delete.  We just need to indicate that this is the first item in the list of values for this key.

Comment (chshu) [inline src/core/common/settings.cpp]: Yes, we don't need the delete action here. I have updated this PR, thanks.

File: src/core/common/settings.cpp [hunk 2]
```diff
@@ -76,6 +76,16 @@ struct settingsBlock
     uint16_t reserved;
 } OT_TOOL_PACKED_END;
 
+static uint16_t getAlignLength(uint16_t length)
+{
+    while (length & 3)
```
Comment (jwhui) [inline src/core/common/settings.cpp]: Sorry, I missed this the first time.  You could simplify this with: `(length + 3) & 0xfffc`

Comment (chshu) [inline src/core/common/settings.cpp]: Done, thanks.

**Pull request merged**